analyzers/src/SonarAnalyzer.CSharp/LiveVariableAnalysis/SonarCSharpLiveVariableAnalysis.cs

```diff
@@ -70,166 +70,175 @@ internal static bool IsLocalScoped(ISymbol symbol, ISymbol declaration)
                 {
-                    state.UsedBeforeAssigned.Add(disposableSymbol);
+                    var foreachNode = (ForEachStatementSyntax)foreachBlock.BranchingNode;
+                    ProcessVariableInForeach(foreachNode);
+                }
+
+                // Keep alive the variables declared and used in the using statement until the UsingFinalizerBlock
+                if (block is UsingEndBlock usingFinalizerBlock)
+                {
+                    var disposableSymbols = usingFinalizerBlock.Identifiers
+                        .Select(x => semanticModel.GetDeclaredSymbol(x.Parent) ?? semanticModel.GetSymbolInfo(x.Parent).Symbol)
+                        .WhereNotNull();
+                    foreach (var disposableSymbol in disposableSymbols)
+                    {
+                        UsedBeforeAssigned.Add(disposableSymbol);
+                    }
                 }
             }
-        }
 
-        private void ProcessVariableInForeach(ForEachStatementSyntax foreachNode, State state)
-        {
-            if (semanticModel.GetDeclaredSymbol(foreachNode) is { } symbol)
+            private void ProcessVariableInForeach(ForEachStatementSyntax foreachNode)
             {
-                state.Assigned.Add(symbol);
-                state.UsedBeforeAssigned.Remove(symbol);
+                if (semanticModel.GetDeclaredSymbol(foreachNode) is { } symbol)
+                {
+                    Assigned.Add(symbol);
+                    UsedBeforeAssigned.Remove(symbol);
+                }
             }
-        }
 
-        private void ProcessVariableDeclarator(VariableDeclaratorSyntax instruction, State state)
-        {
-            if (semanticModel.GetDeclaredSymbol(instruction) is { } symbol)
+            private void ProcessVariableDeclarator(VariableDeclaratorSyntax instruction)
             {
-                state.Assigned.Add(symbol);
-                state.UsedBeforeAssigned.Remove(symbol);
+                if (semanticModel.GetDeclaredSymbol(instruction) is { } symbol)
+                {
+                    Assigned.Add(symbol);
+                    UsedBeforeAssigned.Remove(symbol);
+                }
             }
-        }
 
-        private void ProcessSimpleAssignment(AssignmentExpressionSyntax assignment, SonarState state)
-        {
-            var left = assignment.Left.RemoveParentheses();
-            if (left.IsKind(SyntaxKind.IdentifierName)
-                && semanticModel.GetSymbolInfo(left).Symbol is { } symbol
-                && IsLocalScoped(symbol))
+            private void ProcessSimpleAssignment(AssignmentExpressionSyntax assignment)
             {
-                state.AssignmentLhs.Add(left);
-                state.Assigned.Add(symbol);
-                state.UsedBeforeAssigned.Remove(symbol);
+                var left = assignment.Left.RemoveParentheses();
```

Comment:
Should we put a comment here that we do not support the fields?

Reply:
I can understand that from the `&& IsLocal(symbol)` condition on line 170.